packages/application-shell/src/components/setup-flop-flip-provider/setup-flop-flip-provider.tsx

```diff
@@ -17,6 +17,40 @@ import type {
 } from '../../types/generated/mc';
 import AllFeaturesQuery from './fetch-all-features.mc.graphql';
 
+const WAIT_FOR_REMOTE_FLAGS_TIMEOUT = 500; // milliseconds
+
+type TStatusAwareRendererProps = {
+  shouldWaitForRemoteFlags?: boolean;
+  children: ReactNode;
+};
+const StatusAwareRenderer = (props: TStatusAwareRendererProps) => {
+  const [shouldRender, setShouldRender] = useState(
+    !props.shouldWaitForRemoteFlags
+  );
+  const { isReady } = useAdapterStatus();
+  const displayTimeout = useRef<NodeJS.Timeout>();
+
+  if (isReady && !shouldRender) {
```

Comment:
We take action after remote flags resolution only if the timeout have not happened so we haven't rendered the children yet.